internal/rbd/rbd_util.go

```diff
@@ -2141,18 +2142,23 @@ func (rv *rbdVolume) setAllMetadata(parameters map[string]string) error {
 
 // unsetAllMetadata unset all the metadata from arg keys on RBD image.
 func (rv *rbdVolume) unsetAllMetadata(keys []string) error {
+	var errno syscall.Errno
 	for _, key := range keys {
 		err := rv.RemoveMetadata(key)
-		// TODO: replace string comparison with errno.
-		if err != nil && !strings.Contains(err.Error(), "No such file or directory") {
-			return fmt.Errorf("failed to unset metadata key %q on %q: %w", key, rv, err)
+		if err != nil {
```

Comment:
please add an E2E to ensure this working as expected or please test this manually by removing the metadata in the backend.